tests/ui/consts/issue-70942-trait-vs-impl-mismatch.stderr

```diff
@@ -10,6 +10,12 @@ note: type in trait
 LL |     const VALUE: usize;
    |                  ^^^^^
 
+note: erroneous constant used
+  --> $DIR/issue-70942-trait-vs-impl-mismatch.rs:13:18
+   |
+LL |     let _: [i32; Zero::VALUE] = [];
+   |                  ^^^^^^^^^^^
+
```

Comment:
Hm, I wish I knew where these come from. We do check `!info.is_tainted_by_errors()` before emitting "erroneous constant used". Is something wrong somewhere and the marking-as-tained doesn't work?

Reply:
`new_error` doesn't take a span, so I think the span is still lost there?

Reply:
Honestly I'm not convinced that this is a regression. We now show where the associated const got instantiated, that seems useful.

The `is_tainted_by_errors` still works and suppresses tons of errors elsewhere.

Reply:
Previously we called `eval_mir_constant` which then also used `ctfe_query`. So the use of `ctfe_query` is not what changed the behavior here. We are just getting more spans than before, for some reason.

Reply:
Looking at the code
```rust
trait Nat {
    const VALUE: usize;
}

struct Zero;

impl Nat for Zero {
    const VALUE: i32 = 0;
    //~^ ERROR implemented const `VALUE` has an incompatible type for trait
}

fn main() {
    let _: [i32; Zero::VALUE] = [];
}
```
What I would expect to happen is that `VALUE` gets tainted since it had the wrong type. But there's no concept of tainting a MIR constant, is there, @oli-obk ? That seems to be the underlying problem.

Reply:
You could return a ty::Const of the error variant

Reply:
That can't be entirely right; `self.eval_mir_constant(&ct, Some(span), None)?` does not show the error.

I think it's not the resolve, it's normalization where the error occurs.

Reply:
> it is resolve that fails, not evaluation.

Okay that was the right hint, I got something that fixes the regression. :)